lib/commands/devices/index.ts

```diff
@@ -85,6 +81,7 @@ export default class DevicesCmd extends Command {
 		let devices;
 
 		if (options.application != null) {
+			const { tryAsInteger } = await import('../../utils/validation');
 			devices = (await balena.models.device.getAllByApplication(
 				tryAsInteger(options.application),
```

Comment:
And here... I'll stop pointing out: It looks like all occurrences of `tryAsInteger` in the CLI codebase, when applied to application names, could / should be replaced with `getApplication` in `sdk.ts` so that app names that look like a number could be used.